html/janus.js

```diff
@@ -1429,13 +1438,12 @@ function Janus(gatewayCallbacks) {
 			// If we got here, we're not screensharing
 			if(media === null || media === undefined || media.video !== 'screen') {
 				// Check whether all media sources are actually available or not
-				// as per https://github.com/meetecho/janus-gateway/pull/114
-				MediaStreamTrack.getSources(function(sources) {
-					var audioExist = sources.some(function(source) {
-						return source.kind === 'audio';
+				navigator.mediaDevices.enumerateDevices().then(function(devices) {
```

Comment:
I overrided the adapter.js with the newest one and applied the newer API : https://github.com/webrtc/adapter/blob/master/adapter.js 
Need more test, based on what you support and how would you define the proper fallbacks. But you can find here the basic idea: https://webrtc.github.io/samples/src/content/devices/input-output/